internal/provider/base_eda_datasource.go

```diff
@@ -0,0 +1,198 @@
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required: true,
+			},
+			"url": schema.StringAttribute{
+				Computed: true,
+			},
+			"id": schema.Int64Attribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+// ConfigValidators validates configuration in a declarative way.
+func (d *BaseEdaDataSource) ConfigValidators(_ context.Context) []datasource.ConfigValidator {
+	// You have at least an name
+	return []datasource.ConfigValidator{
+		datasourcevalidator.Any(
+			datasourcevalidator.AtLeastOneOf(
+				tfpath.MatchRoot("name")),
+		),
+	}
+}
+
+// ValidateConfig validates configuration in an imperative way.
+func (d *BaseEdaDataSource) ValidateConfig(ctx context.Context, req datasource.ValidateConfigRequest, resp *datasource.ValidateConfigResponse) {
+	// Check that the response and diagnostics pointer is defined
+	if resp == nil {
+		tflog.Error(ctx, "Response not defined, we cannot continue with the execution")
+		return
+	}
+
+	// Check that the current context is active
+	if !IsContextActive("ValidateConfig", ctx, &resp.Diagnostics) {
+		return
+	}
+
+	var data BaseEdaSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if IsValueProvidedOrPromised(data.Name) {
+		return
+	}
+
+	if !IsValueProvidedOrPromised(data.Name) {
+		resp.Diagnostics.AddAttributeWarning(
+			tfpath.Root("id"),
+			"Missing Attribute Configuration",
+			"Expected [id]",
+		)
+	}
+}
+
+// Configure adds the provider configured client to the data source.
+func (d *BaseEdaDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Check that the response and diagnostics pointer is defined
+	if resp == nil {
+		tflog.Error(ctx, "Response not defined, we cannot continue with the execution")
+		return
+	}
+
+	// Check that the current context is active
+	if !IsContextActive("Configure", ctx, &resp.Diagnostics) {
+		return
+	}
+
+	// Check that the provider data is configured
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*AAPClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *AAPClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	d.client = client
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (d *BaseEdaDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state BaseEdaSourceModel
+	var diags diag.Diagnostics
+
+	// Check Read preconditions
+	if !DoReadPreconditionsMeet(ctx, resp, d.client) {
+		return
+	}
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+
+	// Create the EDA path with query parameters
+	resourceURL := path.Join(d.client.getEdaApiEndpoint(), d.ApiEntitySlug)
```

Comment:
What happens if the EDA is not present? I think `d.client.getEdaApiEndpoint()` will return an empty string and this will probably lead to a 404.

Maybe we could check if `d.client.getEdaApiEndpoint()` is empty and add an error to `resp.Diagnostics` noting that the EDA API is unavailable?

Reply:
Good point. There is nothing preventing someone from attempting to use the provider against an AAP deployment that does not have EDA set up. I'll add a check and error out if the endpoint is empty.

Reply:
I've added the check for empty string!